### DISCUSSION TITLE
Issue with running Hugo using NPM scripts

### DISCUSSION THREAD

**Discussion by hugo-sid** [jmooring/hvm#82] [Announcements]
Hi Joe. First of all, thank you for creating this great tool.

I am using HVM for a Hugo project.

I also have the following in `package.json`:

```json
  "scripts": {
    "dev": "hugo server --navigateToChanged --gc"
  }
```

When I run `npm run dev`, I get the following error:

```
> hugo server --navigateToChanged --gc

sh: 1: hugo: not found
```

However, if I run `hugo server --navigateToChanged --gc` directly in the terminal, it works perfectly fine.

It would be great if HVM worked with `npm run *`, since I will not have to remember lengthy options (like `--navigateToChanged`).

**Comment by jmooring** [comment]
In which file did you place the alias function?

**Comment by hugo-sid** [comment]
`~/.bashrc`

**Comment by jmooring** [comment]
I suspect `npm run foo` uses the Bourne (or some other) shell not Bash, so it doesn't see the aliases. See:
<https://docs.npmjs.com/cli/v9/using-npm/config#script-shell>

**Comment by jmooring** [comment]
Upon further review and research, this may not be possible without creating a separate script that npm can call directly, and that's not something I'm going to spend any time on. Moving this to discussions.

**Comment by hugo-sid** [comment]
Thank you Joe, for your efforts.

**Comment by CyrusYip** [comment]
Since you use npm, I would recommend using [hugo-extended](https://www.npmjs.com/package/hugo-extended) or [hugo-bin](https://www.npmjs.com/package/hugo-bin).

[package.json example](https://github.com/CyrusYip/cyrusyip-blog/blob/474f8a45965c7640bb23b79414acd668e5673e74/package.json).

**Reply by brennanangel** [reply]
This was a huge help! Thank you @CyrusYip
